cetmix_tower_git/models/cx_tower_server.py

```diff
@@ -59,3 +61,49 @@ def _update_or_create_related_record(
         return super()._update_or_create_related_record(
             model, reference, values, create_immediately=create_immediately
         )
+
+    @api.model
+    def get_servers_by_git_ref(self, repository_url, head=None, head_type=None):
+        """
+        Return a recordset of cx.tower.server records related to the specified
+        Git repository URL and reference (branch, commit, or PR/MR).
+
+        Parameters
+        ----------
+        repository_url : str
+            The raw Git remote URL (SSH, HTTPS, git://, etc.).
+        head : str, optional
+            Branch name, commit SHA, or pull/merge request identifier.
+        head_type : {'branch', 'commit', 'pr'}, optional
+            Discriminator for branch, commit, or pull/merge request.
+
+        Returns
+        -------
+        recordset of cx.tower.server
+            All servers matching the given Git reference criteria.
+        """
+        # Normalize incoming URL to a canonical form
+        canonical = normalize_git_url(repository_url)
```

Comment:
We assume that URL provided as an argument is already normalised. So normalisation is not the thing this function should take care of.  Normalisation should be done before this function is even called.

So the task is to:

- Get the URL
- Search by URL
- Return the servers

That is.